optimade/validator/validator.py

```diff
@@ -513,16 +592,24 @@ def test_page_limit(self, response, check_next_link=True):
         except KeyError:
             raise ResponseError("Field `meta->more_data_available` was missing.")
 
-        if more_data_available:
+        if more_data_available and bool(check_next_link):
```

Comment:
```suggestion
        if more_data_available and check_next_link:
```

Reply:
For some reason I just had a vivid flashback, I remember thinking that would you suggest this change when I wrote it the original :P